sky/clouds/cloud.py

```diff
@@ -448,6 +476,44 @@ def check_features_are_supported(
                     f'The following features are not supported by {cls._REPR}:'
                     '\n\t' + table.get_string().replace('\n', '\n\t'))
 
+    @classmethod
+    def get_excludable_features(
+        cls, requested_features: Set[CloudImplementationFeatures],
+        config: ExcludableFeatureCheckConfig
+    ) -> Set[CloudImplementationFeatures]:
+        """Returns the features that can be excluded/ignored by the cloud implementation, given their exclusion condition is met.
+
+        For instance, Kubernetes Cloud can exclude autostop for spot controller, so
+        Kubernetes.check_excludable_features({
+            CloudImplementationFeatures.AUTOSTOP
+        }) returns {CloudImplementationFeatures.AUTOSTOP} if the cluster is a spot controller else {}.
+        """
+        excludable_features = set()
+        excludable_features2condition = cls._cloud_excludable_features()
+        for feature, condition in excludable_features2condition.items():
+            if feature in requested_features and condition(config):
+                excludable_features.add(feature)
+        return excludable_features
+
+    @classmethod
+    def grant_features(
+        cls, requested_features: Set[CloudImplementationFeatures],
+        config: ExcludableFeatureCheckConfig
+    ) -> Set[CloudImplementationFeatures]:
+        """Returns the features that can be granted by the cloud implementation.
```

Comment:
Would be great if adding a comment saying the `cloud.check_features_are_supported` checking is moved to here 🙂

Reply:
Sounds good, I can work on it

Reply:
Updated.